app/src/main/java/org/piramalswasthya/cho/ui/commons/personal_details/PersonalDetailsFragment.kt

```diff
@@ -296,6 +356,194 @@ class PersonalDetailsFragment : Fragment() {
     var pageWidth = 792
 
 
+    private fun checkAndRequestCameraPermission() {
+        if (checkSelfPermission(requireContext(),Manifest.permission.CAMERA) == PackageManager.PERMISSION_GRANTED &&
+            checkSelfPermission(requireContext(),Manifest.permission.WRITE_EXTERNAL_STORAGE
+            )  == PackageManager.PERMISSION_GRANTED
+        ) {
+            // Camera permission is granted, proceed to take a picture
+            takePicture()
+        } else {
+            // Camera permission is not granted, request it
+            requestCameraPermission()
+        }
+    }
+    private fun requestCameraPermission() {
+        val permission = arrayOf<String>(Manifest.permission.CAMERA,Manifest.permission.WRITE_EXTERNAL_STORAGE )
+        requestPermissions(permission, 112)
+    }
+    private fun takePicture() {
+        val photoFile: File? = try {
+            createImageFile()
+        } catch (ex: Exception) {
+            null
+        }
+
+        photoFile?.also {
+            photoURI = FileProvider.getUriForFile(
+                requireContext(),
+                "org.piramalswasthya.cho.provider",
+                it
+            )
+            takePictureLauncher.launch(photoURI)
+
+        }
+    }
+    private fun createImageFile(): File {
+        val timeStamp = SimpleDateFormat("yyyyMMdd_HHmmssSSS", Locale.getDefault()).format(Date())
+        val storageDir = requireContext().getExternalFilesDir("images")
+        currentFileName = "JPEG_${timeStamp}_.jpeg"
+        var file = File(storageDir, currentFileName)
+
+        // Ensure the file doesn't already exist
+        var counter = 1
+        while (file.exists()) {
+            currentFileName = "JPEG_${timeStamp}_$counter.jpeg"
+            file = File(storageDir, currentFileName)
+            counter++
+        }
+
+        return file.apply {
+            // Save a file path for use with ACTION_VIEW intents
+            currentPhotoPath = absolutePath
+        }
+
+    }
+    private val takePictureLauncher =
+        registerForActivityResult(ActivityResultContracts.TakePicture()) { result: Boolean ->
+            if (result) {
+                val highspeed = FaceDetectorOptions.Builder()
+                    .setPerformanceMode(FaceDetectorOptions.PERFORMANCE_MODE_FAST)
+                    .build()
+                val detector = FaceDetection.getClient(highspeed)
+                val image = InputImage.fromFilePath(requireContext(), photoURI)
+                detector.process(image)
+                    .addOnSuccessListener { faces ->
+                        if (faces.isEmpty()) {
+                            Toast.makeText(requireContext(), "No face detected", Toast.LENGTH_SHORT).show()
```

Comment:
harcoded text can be added in string file so if localization use message can be seen in user specific language